consensus/src/consensus/mod.rs

```diff
@@ -842,10 +870,73 @@ impl ConsensusApi for Consensus {
         })
     }
 
+    fn get_block_even_if_header_only(&self, hash: Hash) -> ConsensusResult<Block> {
+        let Some(status) = self.statuses_store.read().get(hash).unwrap_option().filter(|&status| !status.has_block_header()) else {
+            return Err(ConsensusError::BlockNotFound(hash));
+        };
+        Ok(Block {
+            header: self.headers_store.get_header(hash).unwrap(),
+            transactions: if status.is_header_only() { Arc::new(vec![]) } else { self.block_transactions_store.get(hash).unwrap() },
+        })
+    }
+
+    fn get_block_info(&self, hash: Hash) -> ConsensusResult<BlockInfo> {
+        let (exists, block_status) = match self.get_block_status(hash) {
+            Some(status) => (true, status),
+            None => (false, BlockStatus::StatusInvalid),
+        };
+
+        // If the status is invalid, then we don't have the necessary reachability data to check if it's in PruningPoint.Future.
+        if block_status == BlockStatus::StatusInvalid {
+            return Ok(BlockInfo::with_exists(exists));
+        }
+
+        // FIXME: check this call
+        let ghostdag_data = self.ghostdag_store.get_data(hash).unwrap();
+
+        Ok(BlockInfo::new(
+            exists,
+            block_status,
+            ghostdag_data.blue_score,
+            ghostdag_data.blue_work,
+            ghostdag_data.selected_parent,
+            (*ghostdag_data.mergeset_blues).clone(),
+            (*ghostdag_data.mergeset_reds).clone(),
+        ))
+    }
+
+    fn get_block_children(&self, hash: Hash) -> Option<Arc<Vec<Hash>>> {
+        self.relations_stores.read()[0].get_children(hash).unwrap_option()
```

Comment:
it's more strait forward to use `self.relations_service` (it wraps level 0 already)